module/apmgopg/hook.go

```diff
@@ -0,0 +1,72 @@
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build go1.9
+
+package apmgopg
+
+import (
+	"github.com/go-pg/pg"
+
+	"go.elastic.co/apm"
+	"go.elastic.co/apm/module/apmsql"
+)
+
+const elasticApmSpanKey = "go-apm-agent:span"
+
+type QueryHook struct {
+	Database string
+	Username string
+
+	UseFormattedQuery bool
+}
+
+func (qh *QueryHook) BeforeQuery(evt *pg.QueryEvent) {
+	var sql string
+	var err error
+
+	if qh.UseFormattedQuery {
+		sql, err = evt.FormattedQuery()
+	} else {
+		sql, err = evt.UnformattedQuery()
+	}
+
+	if err != nil {
+		// todo: panic/logging ?
```

Comment:
@axw Not sure about this, not a fan of missing the error handling but I don't want to break an application because of a logging failure either.

Reply:
How about just give the span a placeholder name, like "(go-pg)", and leave the statement in the database context empty? I think that's probably better than dropping spans.